packages/ui-react/src/components/nft/components/provider.tsx

```diff
@@ -0,0 +1,45 @@
+import {
+	createContext,
+	useContext,
+	useMemo,
+	type PropsWithChildren,
+} from "react";
+import type { Address } from "viem";
+import type { NFTCardType } from "../types";
+import { useGetSingleNFTMetadata } from "#hooks/data/use-blockscout.js";
+import type { NFTResponse } from "#lib/blockscout/types.js";
+
+export type NFTContextType = {
+	nftData: NFTResponse | undefined;
+	loading: boolean;
+	error: Error | null;
+};
+
+export const NFTContext = createContext<NFTContextType>(undefined as never);
+
+export const NFTProvider = ({
+	children,
+	contractAddress,
+	tokenId,
+}: PropsWithChildren<NFTCardType>) => {
+	const {
```

Comment:
in this case it's fine while generally we might prefer nanostore over context provider 
https://github.com/nanostores/nanostores